src/main/webapp/app/exercises/programming/shared/code-editor/service/code-editor-repository.service.ts

```diff
@@ -168,34 +168,37 @@ export class CodeEditorRepositoryFileService extends DomainDependentEndpointServ
         this.fileUpdateUrl = `${this.restResourceUrl}/files`;
     }
 
-    getRepositoryContent = () => {
-        return this.http.get<{ [fileName: string]: FileType }>(`${this.restResourceUrl}/files`).pipe(handleErrorResponse<{ [fileName: string]: FileType }>(this.conflictService));
+    getRepositoryContent = (domain?: DomainChange) => {
+        const restResourceUrl = domain ? this.calculateRestResourceURL(domain) : this.restResourceUrl;
```

Comment:
This line is repeated frequently. Couldn't we move the logic to fall back to ``this.restResourceUrl`` into ``calculateRestResourceURL``?

Reply:
The idea is that we don't want to calculate the url new every time, only if we get a new domain. So I don't think it makes sense to call this method everytime (in fact it's not possible to call it with undefined). I thought about moving the ` ? : ` chain into a method but thought it's too small and a too comon operation for this. Do you stil think it should be its own method?

Reply:
Could just be as simple as a
```js
if (!domain) {
    return this.restResourceUrl;
}
```
right at the start of ``calculateRestResourceURL``.
But yeah, it works this way, too :)